### DISCUSSION TITLE
Things I Hope Will Be In 2.0 and Beyond

### DISCUSSION THREAD

**Discussion by DWrightAnimation** [keiwando/toonsquid#550] [General]
- Special Effects (such as Gaussian Blur, Motion Blur, etc.)
- Fixed Onion Skin Toggle (We currently only have Relative Onion Skin Toggle) 
- Improvements to the Out of Pegs Feature (Currently, you can only edit it once, and you have to reset it in order to re-edit it, which isn't convenient) 
- Animatable Warp and Perspective 
- Inverse Kinematics (To make animating arms and legs on a rigged character easier)
- More options for the Pen Tool (pressure curves, alternate brush looks other than the basic one, etc.) 
- Animatable Cycles (The only way to do this currently is to make a symbol out of the thing you want to repeat, and while it can allow for it to be animated, it isn't convenient for a one and done thing for hand drawn animation)
- Transitions (For moments where a character moves in slow motion, but it fades from one key pose to another)
- Ability to Stretch Limbs with Inverse Kinematics 
- Keyframe Markers (basically, if you wanted a smile marker to turn into a frown marker, but you have an O-mouth in between the two, this option would make animation between the smile and frown without the O-mouth)
- More Animation Options for Groups (Opacity is currently the only animatable option)

**Comment by giloo33** [comment]
One feature I really missing is key frame countdown. From point a to point b on kayframe, moving up or down number like a countdown effect. 
ToonSquid it's really good for a office presentation etc, nice and easy statistic to explains by diagrams buy number work moving from 1 to 1000 it's not possible yet.

**Reply by DWrightAnimation** [reply]
Do you have a visual example, such as a video?

**Reply by Lioncub901** [reply]
This would be very useful to make timers in ToonSquid. I feel instead of a counter. ToonSquid should allow variables with simple code. Allow scripting.

**Comment by budaloops** [comment]
I'm hyped for 2.0. Wonder if it will be released this year?

**Reply by DWrightAnimation** [reply]
I wonder the same thing, I hope we get it soon enough. I'd love to help get more people to buy the app, so it can eventually be industry standard, and it can help Keiwan with future updates.

**Comment by TempleLionStudios** [comment]
* Exiting symbol editing: lost in timeline
* When animating in symbols, project settings for frame adding (new frame: 2) is not carried over.
* Moving something slightly is hard and often needed. In procreate you select an object, then tab beside it to move it one pixel at a time 
* Onionskin solo on one layer (disable all others)
* Fill within a selection. Selecting an area then fill that area.
* Layer filter adjustments: Gaussian Blur, HSB, opacity (should be under filters instead), idea: chromatic aboration 🤷🏽‍♂️

* Export video w/ transparent background option
* improved fill tool, that doesn’t need constant adjusting (gap-closing/tolerance) to avoid the tiny space between fill and line art or overflooding the canvas. Could AI somehow be used for coloring? 🤷🏽‍♂️ (i know very little about programming, just a thought) 
* Importing symbols from other projects
* Vector lines often bulge out randomly, instead of a stead line with same thickness
* Quickshape makes lines random thickness when in vector, so not the same as a normal drawn line (maybe also pixel?)
* Moving pixels around quickly loses quality. Could we have an option for perfect pixel moving so it doesn’t get blurry?

**Reply by FoxeyeStudios** [reply]
Um… Buddy, A lot of the things you’re asking for is already there… How long have you been using ToonSquid?

**Reply by TempleLionStudios** [reply]
I use it every day and have used it for two years or so.

**Reply by TempleLionStudios** [reply]
> Um… Buddy, A lot of the things you’re asking for is already there… How long have you been using ToonSquid?

Can you name the ones you believe sre already there?

**Reply by Lioncub901** [reply]
Import symbol exist, same as transparent videos, you can put the onion skin on one layer (just turn off the onion skin of the other layers), don’t really understand what you mean by remove detail when transforming (it’s just the same for me. Maybe just turn off anti aliasing),  you can fill in a selected area just tap the fill bucket and then tap selected area, ai for fill tool is a bit much the gap-closing/tolerance works fine imo. I would give you a tutorial, but I have to sleep sorry.

**Reply by TempleLionStudios** [reply]
> Import symbol exist, same as transparent videos, you can put the onion skin on one layer (just turn off the onion skin of the other layers), don’t really understand what you mean by remove detail when transforming (it’s just the same for me. Maybe just turn off anti aliasing),  you can fill in a selected area just tap the fill bucket and then tap selected area, ai for fill tool is a bit much the gap-closing/tolerance works fine imo. I would give you a tutorial, but I have to sleep sorry.

- It seems the only way to import symbols is by going into a project copying then pasting. This is not what I mean. Im looking for a library for each movie project. 

- Transparent video does not work when you import it to other apps. I have tried this multiple times. Ot doesn't seem to work. 

- Having to disable onionskins for 34 layers takes a long time and is not efficient. Soloing ome layer makes sense. You use this in music production often, when soloing tracks. 

- moving graphics lower the quality significantly. You can't really move anything two times or it will look too distorted. 

- filling a selected area doesn't always work. I've had it bug out many times. 

- the gap closer is not the best. it often wont fill even with tiny gaps. Especially with vector line art. When you color everyday you start to notice how many times the tolerance/gap closing is faulty. 

- using ai for coloring is actually genius. Imagine you could color the first frame and the ai could color the entire rest of the scene. Coloring is for many animators the most time consuming and not the most enjoyable part. I myslef do enjoy it, but it takes a looong time. Ai **WILL** be used for this purpose by other apps in near future, and since TS Is my fav app I wish for it to be there too.

**Reply by Lioncub901** [reply]
- watch this: https://www.youtube.com/watch?v=EJi9BJ18eIU
- video below
- try multi selecting and then disable onion skin. The use case of just isolating one is not really ideal since maybe someone might want to isolate 2 or 3 frame, how would they do that?
- I tried doing that and it didn’t really change that much quality for me (maybe disable anti aliasing might fix it)
- works for me (you may be trying to fill in a layer that has a line thus it would cut it off create a new layer and the fill)
- oh, I think gap closing is only for pixel for vector it does not really need that (it only needs the way to cut off intersecting lines which has already been requested).
- I know AI would be a cool feature, but I don’t think it’s that necessary compared to improving the animation experience by adding effects/rigging/ etc that is why I said it was a bit much.

https://github.com/keiwando/toonsquid/assets/88890833/b341a848-63f8-4f3a-96f2-fc2e77cc5f9d

**Comment by CowlBlue** [comment]
I’ve seen some discussion here and there about the gap closing in the fill tool so I figured I’d add my two cents about it:  

from what I can tell, the gap closing seems to work a bit better for drawings with anti-aliased lines, but not very well for aliased line-work.  Aliased line-work is important because images with aliased lines and flat color gives a lot of flexibility for compositing software like after effects to manipulate those images since the colors are really easy to select and modify.

Granted, from what I’ve learned about software development, things like gap closing and automatic shape selection is not an easy thing to do and can be very mathematically tedious to figure out, which is why I don’t particularly mind if things like that are a bit lower priority.  

-That being said, I’m very curious and excited for what 2.0 has in store for us!
